x-pack/plugins/enterprise_search/server/lib/ml_inference_pipeline/get_inference_pipelines.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IngestPipeline, IngestProcessorContainer } from '@elastic/elasticsearch/lib/api/types';
+import { ElasticsearchClient } from '@kbn/core/server';
+import { MlTrainedModels } from '@kbn/ml-plugin/server';
+
+/**
+ * Gets all ML inference pipelines. Redacts trained model IDs in those pipelines which reference
+ * a model inaccessible in the current Kibana space.
+ * @param esClient the Elasticsearch Client to use to fetch the errors.
+ * @param trainedModelsProvider ML trained models provider.
+ */
+export const getMlInferencePipelines = async (
+  esClient: ElasticsearchClient,
+  trainedModelsProvider: MlTrainedModels | undefined
+): Promise<Record<string, IngestPipeline>> => {
+  if (!trainedModelsProvider) {
+    return Promise.reject(new Error('Machine Learning is not enabled'));
+  }
+
+  // Fetch all ML inference pipelines and trained models that are accessible in the current
+  // Kibana space
+  const [fetchedInferencePipelines, trainedModels] = await Promise.all([
+    esClient.ingest.getPipeline({
+      id: 'ml-inference-*',
+    }),
+    trainedModelsProvider.getTrainedModels({}),
+  ]);
+  const accessibleModelIds = Object.values(trainedModels.trained_model_configs).map(
+    (modelConfig) => modelConfig.model_id
+  );
+
+  // Process pipelines: check if the model_id is one of the redacted ones, if so, redact it in the
+  // result as well
+  const inferencePipelinesResult: Record<string, IngestPipeline> = {};
+  Object.entries(fetchedInferencePipelines).forEach(([name, inferencePipeline]) => {
```

Comment:
Again for future reference, but `forEach` should generally be avoided. It's inefficient and has some limitations that traditional for loops don't have. If you find yourself needing to do for loops (which, also for future reference, you generally shouldn't want to do), just construct a traditional for loop.

In this case, using `reduce` to collate the data in a single object would probably have been a better choice and more idiomatic for our codebase.

Reply:
Thanks for the tip! Addressed in #143967.